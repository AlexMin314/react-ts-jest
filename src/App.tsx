import * as React from 'react';
import './App.css';
import {TodoList} from './components/TodoList';
import {TodoStore} from './stores/TodoStore';
import {inject, observer} from 'mobx-react';
import {TodoInput} from './components/TodoInput';

const logo = require('./logo.svg');

interface AppProps {
    todoStore?: TodoStore;
}

@inject('todoStore')
@observer
class App extends React.Component<AppProps, {}> {
    render() {
        return (
            <div className="App">
                <div className="App-header">
                    <img src={logo} className="App-logo" alt="logo"/>
                    <h2>Welcome to React</h2>
                </div>
                <TodoInput />
                <TodoList />
            </div>
        );
    }
}

export default App;
