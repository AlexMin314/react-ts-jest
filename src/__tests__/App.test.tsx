import * as React from 'react';
import App from '../App';
import {createRenderer} from 'react-test-renderer/shallow';
import {TodoStore} from '../stores/TodoStore';

describe('App', () => {
    it('renders without crashing', () => {
        // Given
        const todoStore = new TodoStore();
        todoStore.addTodo('hello');
        const renderer = createRenderer();

        // When
        renderer.render(<App todoStore={todoStore}/>);

        // Then
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
});
