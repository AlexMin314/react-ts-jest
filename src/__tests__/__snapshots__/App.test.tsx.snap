// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<App
  todoStore={
    TodoStore {
      "todos": Array [
        Todo {
          "done": false,
          "id": "id-1",
          "label": "hello",
        },
      ],
    }
  }
/>
`;
